tests/failpoints/cases/test_unsafe_recovery.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::iter::FromIterator;
+use std::sync::{Arc, Condvar, Mutex};
+
+use futures::executor::block_on;
+use pd_client::PdClient;
+use raftstore::store::util::find_peer;
+use test_raftstore::*;
+
+#[test]
+fn test_unsafe_recover_send_report() {
+    let mut cluster = new_server_cluster(0, 3);
+    cluster.run();
+    let nodes = Vec::from_iter(cluster.get_node_ids());
+    assert_eq!(nodes.len(), 3);
+
+    let pd_client = Arc::clone(&cluster.pd_client);
+    pd_client.disable_default_operator();
+    let region = block_on(pd_client.get_region_by_id(1)).unwrap().unwrap();
+    configure_for_lease_read(&mut cluster, None, None);
+
+    let store2_peer = find_peer(&region, nodes[1]).unwrap().to_owned();
+    cluster.must_transfer_leader(region.get_id(), store2_peer);
+    cluster.put(b"random_key1", b"random_val1").unwrap();
+
+    let apply_triggered_pair = Arc::new((Mutex::new(false), Condvar::new()));
+    let apply_triggered_pair2 = Arc::clone(&apply_triggered_pair);
+    let apply_released_pair = Arc::new((Mutex::new(false), Condvar::new()));
+    let apply_released_pair2 = Arc::clone(&apply_released_pair);
+    fail::cfg_callback("on_handle_apply_store_1", move || {
+        {
+            let (lock, cvar) = &*apply_triggered_pair2;
+            let mut triggered = lock.lock().unwrap();
+            *triggered = true;
+            cvar.notify_one();
```

Comment:
seems a sync channel is enough?

Reply:
I did a little bit investigation. Seems to me that, `sync channel` does not work here, because the failpoint callback may be called multiple times, and `apply-triggered sender` cannot send messages multiple times while `apply-triggered receiver` only receive once, `apply-released receiver` will be blocked forever since `apply-released sender` only sends once. Correct me if I am wrong

Reply:
seems you just want to block the apply process, why not use `fail::cfg("on_handle_apply_store_1", "pause");`?

Reply:
We can definitely use `pause` to replace the functionality of `apply-released` switch. But, we still need `apply-triggered` switch to ensure that the "reporting" logic happens exactly after an entry is **committed** but **not applied** on this peer.  I combined these 2 requirements in this callback, instead of using a callback for `apply-released` plus a `fail point pause`.